src/ralph/backends/data/async_mongo.py

```diff
@@ -29,22 +29,25 @@
 )
 
 logger = logging.getLogger(__name__)
+Settings = TypeVar("Settings", bound=MongoDataBackendSettings)
 
 
-class AsyncMongoDataBackend(BaseAsyncDataBackend, AsyncWritable, AsyncListable):
+class AsyncMongoDataBackend(
+    BaseAsyncDataBackend[Settings, MongoQuery],
```

Comment:
It might be easier to inherit directly from
```python
BaseAsyncDataBackend[MongoDataBackendSettings, MongoQuery]
```
However, then there is a type issue with `AsyncMongoLRSBackend`  that extends `AsyncMongoDataBackend` and, at the same time, changes its settings to `MongoLRSBackendSettings`.
Thus, using a `Settings` TypeVar here allows future specialization of this class.
Should we apply this pattern to other backends even when we don't specialize them further? And what about allowing the same thing for the `Query` type?
WDYT?

Reply:
I'm not a fan of having to define a `Settings` TypeVar for each backend.
If `lrs` backends would not have proper settings (e.g. they would only inherit from their `data` backend's settings), we wouldn't have to allow future specialization, right?
Only the `fs` backend has a proper `LRS` setting. Is it still relevant for `lrs` backends to have their proper settings class?

Reply:
You are right) If `lrs` backends don't redefine their `settings` we could be less flexible and re-use their `data`  settings equivalent. However, currently, all `lrs` backends redefine their settings to update the `env_prefix` and besides the `fs` backend, the `clickhouse` backend too has one additional lrs-specific setting: `IDS_CHUNK_SIZE`.